io-functions-pay-portal/PostTransactionsChallenge3ds2/index.ts

```diff
@@ -0,0 +1,50 @@
+import * as express from "express";
+import * as winston from "winston";
+
+import { Context } from "@azure/functions";
+
+import { secureExpressApp } from "io-functions-commons/dist/src/utils/express";
+import { AzureContextTransport } from "io-functions-commons/dist/src/utils/logging";
+import { setAppContext } from "io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import createAzureFunctionHandler from "io-functions-express/dist/src/createAzureFunctionsHandler";
+import { getConfigOrThrow } from "../utils/config";
+import { challengePage } from "../utils/page3ds";
+
+const config = getConfigOrThrow();
+
+// tslint:disable-next-line: no-let
+let logger: Context["log"] | undefined;
+const contextTransport = new AzureContextTransport(() => logger, {
+  level: "debug"
+});
+winston.add(contextTransport);
+
+// Setup Express
+const app = express();
+
+secureExpressApp(app);
+
+// Add express route
+
+app.post("/api/v1/transactions/:id/challenge", (req, res) => {
+  res.set("Content-Type", "text/html");
+  return res.send(
+    challengePage
+      .replace(
+        "IO_PAY_CHALLENGE_RESUME_URL",
+        config.IO_PAY_CHALLENGE_RESUME_URL
```

Comment:
as above , we should add it to infra `prod/westeurope/internal/api/functions_iopayportal/function_app/terragrunt.hcl` used var